tls/s2n_crl.h

```diff
@@ -0,0 +1,30 @@
+/*
+* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#pragma once
+
+#include "api/s2n.h"
+
+#include <openssl/x509v3.h>
+
+struct s2n_x509_crl {
+    X509_CRL *crl;
+};
+
+/* TODO: APIs are part of an unfinished CRL validation feature and are temporarily hidden
+ * https://github.com/aws/s2n-tls/issues/3499 */
+int s2n_x509_crl_from_pem(uint8_t *pem, size_t len, struct s2n_x509_crl **crl);
```

Comment:
crl should probably be the first argument in crl methods.

```suggestion
 * https://github.com/aws/s2n-tls/issues/3499 */
int s2n_x509_crl_from_pem(struct s2n_x509_crl **crl, uint8_t *pem, size_t len);
```
I'd also suggest a name that makes it clear that the method will allocate a crl. Maybe just "s2n_x509_new", or "s2n_x509_new_from_pem", or "s2n_x509_alloc_from_pem".

Alternatively, you could split out the _new method so that it's even clearer to customers that they're allocating a new crl. So then you'd have:
```suggestion
 * https://github.com/aws/s2n-tls/issues/3499 */
int s2n_x509_new(struct s2n_x509_crl **crl);
// or "struct s2n_x509_crl *s2n_x509_new()" since new methods might be an exception to our return value rules
int s2n_x509_load_pem(struct s2n_x509_crl *crl); // this matches s2n_cert_chain_and_key_load_pem's naming
```

Reply:
I changed both of these. `crl` is the first argument now, and I made the `s2n_crl_new` function public, and called the other one `s2n_crl_load_pem`.